Pagination header

om-sharma opened the pull request:
Restyled page navigation links on records and entries page.

![screenshot 2015-12-04 12 05 56](https://cloud.githubusercontent.com/assets/8011952/11592986/bb785f1a-9a99-11e5-9cd6-ca36fe95a56d.png)
![screenshot 2015-12-04 12 06 40](https://cloud.githubusercontent.com/assets/8011952/11592993/be618d14-9a99-11e5-9729-1028a39fb3fd.png)
![screenshot 2015-12-04 12 07 31](https://cloud.githubusercontent.com/assets/8011952/11592997/c0a7018a-9a99-11e5-87f5-175e023d9ede.png)

om-sharma committed:
rendering page navigation header and restyled page navigation footer

om-sharma committed:
Rendering "xxx records" when first page renders all the entries

philandstuff commented on src/main/java/uk/gov/register/presentation/resource/Pagination.java:
```diff
@@ -43,6 +43,23 @@ long offset() {
+        return totalEntries;
+    }
+
+    public boolean isFirstPageAndShowsAllEntries() {
```
could this have a better name? `isOnlyPage` or `isSolePage` or something like that?

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+.previous-next-navigation {
```
Can you document where this came from?

philandstuff replied:
maybe steal from https://github.com/alphagov/static/blob/master/app/assets/stylesheets/govuk-component/_previous-and-next-navigation.scss ?

philandstuff replied:
spoke to @dsingleton who said that pulling things in from govuk-component might not be the best idea as it's very specific to GOV.UK's own needs.

If there were a navigation widget in govuk-element it might have been nice to use it, but there isn't.

So, this is fine for now.

ping @robinwhittleton who might be interested in this discussion

dsingleton replied:
Hi, had a look at this PR in a bit more detail after our discussion.

I think re-using the naming convention (`<div class="govuk-previous-and-next-navigation">`) would be a bad idea, as it might suggest a direct link between the code-bases, where one doesn't exists, introducing confusion. Changing the root scope/class name, and keeping the rest of the markup/CSS, would be :+1:, possible with a "based on <git link> comment".

I'd recommend using the component markup/CSS over other GOV.UK implementations as it's likely to be more modern/polished (for example, the component has better accessibility support). That would also be the best starting point for adding to elements.

philandstuff commented:
:+1: other than the minor comments above

om-sharma committed:
Renamed Pagination class method to a much better name

robinwhittleton commented on src/main/resources/templates/records.html:
```diff
@@ -7,9 +7,7 @@
     <div th:replace="main.html::phase"></div>
     <div class="grid-row">
-        <div class="column-two-thirds">
-            <h1 th:text="${entries.size()} + ' records'"></h1>
```
We’re losing our `h1` for the page here. Is there something replacing it?

om-sharma replied:
Yes `h1` is replaced by text i.e. `1-100 of 1000 records` or `56 records`. Have a look at the attached png's with the PR.

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+        width: 90%;
+    }
+
+    p.download {
```
this isn't needed.

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+}
+
+@media only screen and  (max-width: 64em) {
+    .previous-next-navigation {
```
can you add a screenshot demonstrating this bit?

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+		position: static;
+	}
+}
+@media (min-width: 641px) and (max-width: 640px) {
```
what does this media query mean?

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+    padding-left: 0;
+}
+
+@media only screen and  (max-width: 64em) {
```
why is this `64em` rather than `640px` elsewhere in this file?

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+    font-size: 18px;
+    line-height: 1.11111;
+}
+}.previous-next-navigation li.next {
```
the formatting's a bit poor here

philandstuff commented on src/main/sass/_page-navigation.scss:
```diff
@@ -0,0 +1,105 @@
+.previous-next-navigation li.next a {
+    padding: 15px 35px 15px 0;
+}
+.previous-next-navigation li.next a:before {
```
it looks like there's a lot of opportunity to take advantage of sass's nesting features here.

philandstuff committed:
use pagination code from govuk-component

As per @dsingleton's suggestion, this pulls in the
previous-next-navigation code from alphagov/static's govuk-component
section, but renames the toplevel css class to not be something
govuk-specific.
